### PR TITLE
[snappy] Initial integration

### PR DESCRIPTION
**Pull request by AdamKorcz**
This PR adds initial integration of [snappy](https://github.com/google/snappy).

For maintainers of Snappy:
@pwnall

Following integration, oss-fuzz will run Snappys fuzzers continuously. If a bug is discovered, all maintainers get notified by email with a link to a detailed bug report.
To complete this integration, the email addresses of maintainers to receive bug reports are to be added to the contact list, and for that I kindly ask you to write those email addresses in this PR and I will then add them to the list. Note that I have added my own email address to keep an eye out for any breakages, and this means that I will be able to see all bugs including security critical bugs. If you are opposed to having me on the list for a few weeks, please let me know and I will remove myself.

Integration into oss-fuzz is free. Please pay notice to [the bug disclosure policy of oss-fuzz](https://google.github.io/oss-fuzz/getting-started/bug-disclosure-guidelines/).

This PR therefore provides the final step of the work that was started in https://github.com/google/snappy/pull/78 to setup continuous fuzzing of Snappy.

### CODE REVIEW

File: projects/snappy/project.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+homepage: "https://github.com/google/snappy"
+language: c++
+primary_contact: ""
+auto_ccs:
+  - "Adam@adalogics.com"
+sanitizers:
+  - address
+fuzzing_engines:
+  - libfuzzer
```
Comment (inferno-chromium) [inline projects/snappy/project.yaml:9]: Can we enable other fuzzing engines and sanitizers. undefined, ; afl and honggfuzz ?

File: projects/snappy/build.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+cd $SRC/snappy
+mkdir build
+cd build && cmake -DSNAPPY_FUZZING_BUILD=ON -DSNAPPY_BUILD_TESTS=OFF ../ && make
+
+cp *_fuzzer $OUT/
```
Comment (inferno-chromium) [inline projects/snappy/build.sh:22]: missing newline.

**Review by inferno-chromium — APPROVED** [review]
@pwnall - do you want to have any additional contacts that would be cced on bugs, otherwise we can just add costan [at] google

**Comment by inferno-chromium** [comment]
@pwnall @jyrki @qrczak for review (based on @ghemawat recommendation)

File: projects/snappy/project.yaml (hunk 1 above)
Comment (AdamKorcz) [reply projects/snappy/project.yaml:9]: Yes, that should not be a problem. The reason for the current set up is that changes are needed upstream in the build system to support that. Once the maintainers confirm interest in oss-fuzz integration, I will sort it out.

**Review by inferno-chromium — APPROVED** [review]
Straighforward one, merging. Victor also confirmed as point of contact over email.

**Pull request merged**